### PR TITLE
UCS/ARBITER: Fix group rescheduling to another arbiter from dispatch

### LINKED ISSUE #5382: [mtt] arbiter.c:236  Assertion `(group)->arbiter == (arbiter)' failed
Configuration
```
OMPI: 4.0.4rc3
MOFED: MLNX_OFED_LINUX-5.0-1.0.0.0
Module: hpcx-gcc-debug (2020-07-09)
Test module: mtt-tests/hpcx-gcc
Nodes: jazz x2 (ppn=28(x2), nodelist=jazz[03-04])
```
 
MTT log:
 
http://hpcweb.lab.mtl.com/hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20200709_181522_30359_34300_jazz03.swx.labs.mlnx/html/test_stdout_JM3gZv.txt
 
Cmd:
`/hpc/local/benchmarks/daily/next/2020-07-09/hpcx-gcc-redhat7.6/ompi/bin/mpirun -np 32 --display-map -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' -x UCX_ERROR_SIGNALS=SIGILL,SIGSEGV,SIGBUS,SIGFPE,SIGABRT -x UCX_HANDLE_ERRORS=bt -x UCX_IB_PCI_RELAXED_ORDERING=off --bind-to socket -x UCX_NET_DEVICES=mlx5_2:1 -mca osc ucx -x UCX_IB_REG_METHODS=rcache,direct -x UCX_TLS=dc_x,self,sm -mca pmix_base_async_modex 0 -mca mpi_add_procs_cutoff 100000 -mca pmix_base_collect_data 1 -x UCX_IB_NUM_PATHS=4 -x UCX_MAX_EAGER_LANES=4 -x UCX_MAX_RNDV_LANES=4 --map-by node -x UCX_RC_MAX_GET_ZCOPY=8192 -x UCX_RC_TX_NUM_GET_OPS=64 /hpc/local/benchmarks/daily/next/2020-07-09/hpcx-gcc-redhat7.6/ompi/tests/osu-micro-benchmarks-5.6.2/osu_allgatherv`
 
Output:
```sh
Fri Jul 10 01:13:30 2020[1,22]<stderr>:[jazz03:140564:0:140564]     arbiter.c:236  Assertion `(group)->arbiter == (arbiter)' failed
Fri Jul 10 01:13:31 2020[1,22]<stderr>:==== backtrace (tid: 140564) ====
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 0 0x000000000005b0e8 ucs_debug_print_backtrace()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/ucs/debug/debug.c:656
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 1 0x000000000004f402 ucs_arbiter_group_desched_nonempty()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/ucs/datastruct/arbiter.c:236
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 2 0x000000000006b59b ucs_arbiter_group_desched()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/ucs/datastruct/arbiter.h:316
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 3 0x000000000006b59b uct_dc_mlx5_iface_dci_put()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/uct/ib/dc/dc_mlx5_ep.h:398
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 4 0x000000000006b59b uct_dc_mlx5_poll_tx()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/uct/ib/dc/dc_mlx5.c:230
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 5 0x000000000006b59b uct_dc_mlx5_iface_progress()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/uct/ib/dc/dc_mlx5.c:248
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 6 0x000000000002ee6a ucs_callbackq_dispatch()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/ucs/datastruct/callbackq.h:211
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 7 0x000000000002ee6a uct_worker_progress()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/uct/api/uct.h:2346
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 8 0x000000000002ee6a ucp_worker_progress()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/ucp/core/ucp_worker.c:2040
Fri Jul 10 01:13:31 2020[1,22]<stderr>: 9 0x00000000000365dc opal_progress()  /build-result/src/hpcx-gcc-redhat7.6/ompi-v4.0.x/opal/runtime/opal_progress.c:231
Fri Jul 10 01:13:31 2020[1,22]<stderr>:10 0x000000000004c8b5 ompi_request_wait_completion()  /build-result/src/hpcx-gcc-redhat7.6/ompi-v4.0.x/ompi/../ompi/request/request.h:415
Fri Jul 10 01:13:31 2020[1,22]<stderr>:11 0x000000000004c8b5 ompi_request_default_wait()  /build-result/src/hpcx-gcc-redhat7.6/ompi-v4.0.x/ompi/request/req_wait.c:42
Fri Jul 10 01:13:31 2020[1,22]<stderr>:12 0x0000000000087659 ompi_coll_base_sendrecv_actual()  /build-result/src/hpcx-gcc-redhat7.6/ompi-v4.0.x/ompi/mca/coll/base/coll_base_util.c:60
Fri Jul 10 01:13:31 2020[1,22]<stderr>:13 0x0000000000086c0f ompi_coll_base_sendrecv()  /build-result/src/hpcx-gcc-redhat7.6/ompi-v4.0.x/ompi/mca/coll/base/coll_base_util.h:98
Fri Jul 10 01:13:31 2020[1,22]<stderr>:14 0x000000000005eae1 PMPI_Allgatherv()  /build-result/src/hpcx-gcc-redhat7.6/ompi-v4.0.x/ompi/mpi/c/profile/pallgatherv.c:146
Fri Jul 10 01:13:31 2020[1,22]<stderr>:15 0x0000000000401833 main()  /build-result/src/hpcx-gcc-redhat7.6/ompi-v4.0.x/mpi_tests/mpitests/osu-micro-benchmarks-5.6.2/mpi/collective/osu_allgatherv.c:117
Fri Jul 10 01:13:31 2020[1,22]<stderr>:16 0x00000000000223d5 __libc_start_main()  ???:0
Fri Jul 10 01:13:31 2020[1,22]<stderr>:17 0x0000000000401c35 _start()  ???:0
Fri Jul 10 01:13:31 2020[1,22]<stderr>:=================================
Fri Jul 10 01:13:31 2020[1,22]<stderr>:[jazz03:140564] *** Process received signal ***
Fri Jul 10 01:13:31 2020[1,22]<stderr>:[jazz03:140564] Signal: Aborted (6)
Fri Jul 10 01:13:31 2020[1,22]<stderr>:[jazz03:140564] Signal code:  (-6)
Fri Jul 10 01:13:31 2020[1,22]<stderr>:[jazz03:140564] [ 0] Fri Jul 10 01:13:31 2020[1,22]<stderr>:/lib64/libpthread.so.0(+0xf5d0)[0x7f33615b25d0]
Fri Jul 10 01:13:31 2020[1,22]<stderr>:[jazz03:140564] [ 1] Fri Jul 10 01:13:31 2020[1,22]<stderr>:/lib64/libc.so.6(gsignal+0x37)[0x7f336120c207]
Fri Jul 10 01:13:31 2020[1,22]<stderr>:[jazz03:140564] [ 2] Fri Jul 10 01:13:31 2020[1,22]<stderr>:/lib64/libc.so.6(abort+0x148)[0x7f336120d8f8]
```

### PR DESCRIPTION
**Pull request by brminich**
## What
Fix arbiter when the group, while being dispatched on the arbiter, is rescheduled to the another arbiter.

## Why ?
Fixes #5382

### CODE REVIEW

File: src/ucs/datastruct/arbiter.h [hunk 1]
```diff
@@ -113,7 +113,9 @@ typedef enum {
 #define UCS_ARBITER_GROUP_ARBITER_SET(_group, _arbiter) \
     (_group)->arbiter = (_arbiter)
 #define UCS_ARBITER_GROUP_ARBITER_CHECK(_group, _arbiter) \
-    ucs_assert((_group)->arbiter == (_arbiter))
+    ucs_assertv((_group)->arbiter == (_arbiter), \
```
Comment (hoopoepg) [inline src/ucs/datastruct/arbiter.h:116]: minor: ARBITER_CHECK macro may be moved out of scope ```#if ENABLE_ASSERT```

Comment (brminich) [reply src/ucs/datastruct/arbiter.h:116]: why?

Comment (hoopoepg) [reply src/ucs/datastruct/arbiter.h:116]: there is used ucs_assertXXX macro which is disabled when ENABLE_ASSERT is empty

File: src/ucs/datastruct/arbiter.h [hunk 2]
```diff
@@ -113,7 +113,9 @@ typedef enum {
...
+                "%p == %p", (_group)->arbiter, _group)
+#define UCS_ARBITER_GROUP_ARBITER(_group) (_group)->arbiter
```
Comment (yosefe) [inline src/ucs/datastruct/arbiter.h:118]: why do we need this macro UCS_ARBITER_GROUP_ARBITER() ?

Comment (brminich) [reply src/ucs/datastruct/arbiter.h:118]: to take arbiter from dummy group and assign it to the currently dispatched group

Comment (yosefe) [reply src/ucs/datastruct/arbiter.h:118]: i'm pretty sure we can pass it directly w/o this macro

File: src/ucs/datastruct/arbiter.h (hunk 1 above)
Comment (brminich) [reply src/ucs/datastruct/arbiter.h:116]: ok, ic

**Comment by brminich** [comment]
@yosefe, pick to 1.9?

**Comment by yosefe** [comment]
> @yosefe, pick to 1.9?

yes

**Pull request merged**